### PR TITLE
Revisited (AO-13011)

### PR DESCRIPTION
**Pull request by lootek**
JIRA: AO-13011

This is a PR for Integration/feature branch merge

### CODE REVIEW

**Comment by lootek** [comment]
See https://github.com/librato/host-agent-deployment/pull/76 after merging!

File: README.md [hunk 1]
```diff
@@ -1,61 +1,55 @@
...
-      * `hostname` (from scoutd config)
-      * `agent_ruby_bin` (path to scout client binary)
-      * `roles` (using API call to PSM)
-      * `key_processes` (using API call to PSM, comma-separated words list)
-      * `plugin_directory` (always /opt/appoptics/opt/psm)
-    * `/opt/appoptics/etc/config.yaml`
-      * `token`
-      * `hostname_alias` (if hostname was set in scoutd.yml config)
+skip Snap Agent configuration 
+Note: When choosing this option, you need to manually configure the Snap PSM collector plugin. For the detailed configuration information, please refer to https://docs.appoptics.com/kb/host_infrastructure/integrations/psm/
```
Comment (adamiklukasz) [inline README.md]: Link not found.
Ignore if it's on dev branch.

File: README.md [hunk 2]
```diff
@@ -1,61 +1,55 @@
...
+
+## Metrics receiving
+
+You need to enable collector(s) on AppOpctics website:
+* PSM plugins/scripts: https://my.appoptics.com/infrastructure/integrations/psm
```
Comment (adamiklukasz) [inline README.md]: At this point AO says: plugin not found.
Is documentation merge aligned with AO update?

**Comment by tomekkolo** [comment]
This should be merged when aoagent is released.

File: README.md (hunk 1 above)
Comment (lootek) [reply README.md]: it is, ignoring ;P

File: README.md (hunk 2 above)
Comment (lootek) [reply README.md]: yes, it is. This PR will be merged when the corresponding changes in other repos are and the new agent is released

File: lib/snap/snap.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require_relative "config"
+
+class Snap
+  class << self
+    def stop
+      system "service swisnapd stop"
```
Comment (gortyy) [inline lib/snap/snap.rb]: `swisnapd` should be class field

File: lib/snap/config.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  attr_reader :agent_ruby_bin, :ruby_path, :plugin_directory, :agent_data_file, :environment, :hostname, :roles
+
+  def initialize
+    scout = Scout.new
+    @environment = scout.environment
+    @hostname = scout.hostname
+    @roles = scout.roles.join(" ")
+
+    scout_location = %x(find / -iname scout-client 2>/dev/null | head -1).chomp
```
Comment (gortyy) [inline lib/snap/config.rb]: remove this pipe to head and use
```shell
find / -iname scout-client -print -quit 2>/dev/null
```

Comment (gortyy) [reply lib/snap/config.rb]: btw what's happening when nothing's found?

File: lib/snap/snap.rb (hunk 3 above)
Comment (lootek) [reply lib/snap/snap.rb]: fixed in https://github.com/pingdomserver/scout-ao/pull/12

File: lib/snap/config.rb (hunk 4 above)
Comment (lootek) [reply lib/snap/config.rb]: nice one :)

fixed in https://github.com/pingdomserver/scout-ao/pull/12

**Pull request merged**